code/lib/preview-api/src/modules/preview-web/Preview.tsx

```diff
@@ -160,9 +162,23 @@ export class Preview<TRenderer extends Renderer> {
   }
 
   // If initialization gets as far as project annotations, this function runs.
-  async initializeWithProjectAnnotations(projectAnnotations: ProjectAnnotations<TRenderer>) {
-    this.projectAnnotationsBeforeInitialization = projectAnnotations;
+  async initializeWithProjectAnnotations(
+    projectAnnotations: ProjectAnnotations<TRenderer>,
+    storyStore?: StoryStore<TRenderer>
+  ) {
     try {
+      await this.beforeAllCleanup?.();
+      this.beforeAllCleanup = await projectAnnotations.beforeAll?.();
+    } catch (err) {
+      this.renderPreviewEntryError('Error in beforeAll hook:', err as Error);
```

Comment:
How does such a preview entry error looks like?

Reply:
Whatever random thing you might be doing in your beforeAll hook. There's no telling what random shenanigans people may end up (ab)using this for.

Reply:
Okay fair enough